### PR TITLE
token arg and gha updates

### LINKED ISSUE #131: nflverse_releases() and nflverse_download() should accept a token argument
This would make things easier to test, since the tests keep crashing into an API wall right now.

### PR DESCRIPTION
**Pull request by tanho63**
Resolves #131

### CODE REVIEW

File: R/utils_download_nflverse.R [hunk 1]
```diff
@@ -28,10 +29,13 @@
 nflverse_download <- function(...,
                               folder_path = getOption("nflreadr.download_path", default = "."),
                               file_type = getOption("nflreadr.prefer", default = "rds"),
-                              use_hive = file_type %in% c("parquet","csv")
+                              use_hive = file_type %in% c("parquet","csv"),
+                              .token = "default"
```
Comment (mrcaseb) [inline R/utils_download_nflverse.R:33]: Why not directly `.token = gh::gh_token()` instead of the extra work through "default"?

**Review by mrcaseb — COMMENTED** [review]
Looks good to me. Just a question out of curiosity.

File: R/utils_download_nflverse.R (hunk 1 above)
Comment (tanho63) [reply R/utils_download_nflverse.R:33]: I didn't want to add a new import, just a suggests, so handling it after the `check_installed()` step

Comment (mrcaseb) [reply R/utils_download_nflverse.R:33]: Makes sense, thanks for clarifying.

**Review by mrcaseb — APPROVED** [review]

**Pull request merged**